ANT+_Android_SDK/Sample/ANT+ Plugin Sampler/source/src/main/java/com/dsi/ant/antplus/pluginsampler/pulsezones/Dialog_DataInput.java

```diff
@@ -56,12 +63,18 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
         detailsView.findViewById(R.id.radioButton_Zone4).setOnClickListener((View view) -> selectedIdZone = 4);
         detailsView.findViewById(R.id.radioButton_Zone5).setOnClickListener((View view) -> selectedIdZone = 5);
 
-        // Add action buttons
-        builder.setPositiveButton("OK", (DialogInterface dialog, int which) -> {});
 
-                builder.setNegativeButton("Cancel", (dialog, which) -> {
-                    //Let dialog dismiss
-                });
+        //Get shared preferences
+        myPrefs = detailsView.getContext().getSharedPreferences("prefID", Context.MODE_PRIVATE);
+        int userSex = myPrefs.getInt("userSexKey", R.id.radioButton_Female);
```

Comment:
for those attributes like I would create in  the Preferences class an Enum with them. This makes it much clearer if someone else works on your code, what to use and how to use. If you go one level further, you can even hide all these details and just have something like: myPrefs.getAge() for instance.